cupy/lib/_routines_poly.py

```diff
@@ -71,3 +73,48 @@ def polysub(a1, a2):
         out = out.astype(cupy.result_type(a1, a2), copy=False)
         out -= 2 * out - a1
     return out
+
+
+def roots(p):
+    """Computes the roots of a polynomial with given coefficients.
+
+    Args:
+        p (cupy.ndarray): polynomial coefficients.
```

Comment:
Why not support for poly1d inputs?

Reply:
pol1d inputs are not documented in numpy's documentation

Reply:
NumPy's documentation says that the parameters should be `rank-1 array`. So please support it.